recipes/mono/mono.inc

```diff
@@ -0,0 +1,187 @@
+
+# comments to configure flags:
+#   tls (thread local storage) fails to build with pthread
+#   static_mono=no fails to build
+EXTRA_OECONF = " \
+    --enable-shared \
+    --disable-rpath \
+    --disable-system-aot \
+    --with-gnu-ld \
+    --with-tls=__thread \
+    --with-static_mono=yes \
+    --with-shared_mono=yes \
+    --with-xen_opt=no \
+    --with-mcs-docs=no \
+    --with-sigaltstack=yes \
+    ${MONO_X} \
+"
+EXTRA_OECONF:>native = " --disable-boehm"
+EXTRA_OECONF:>machine = " --enable-mcs-build"
+
+# the cross build should use the native version of the
+# 'jay' binary, so fix the makefiles
+DO_CONFIGURE_JAY = ""
+DO_CONFIGURE_JAY:machine = "do_configure_jay"
+do_configure[postfuncs] += "${DO_CONFIGURE_JAY}"
+do_configure_jay() {
+    find ${S} -name Makefile -exec sed -ir 's![^ \t]*jay/jay!jay!' {} \;
+}
+
+# fix cross build calls to mono
+DO_CONFIGURE_LIBTOOL = ""
+DO_CONFIGURE_LIBTOOL:machine = "do_configure_libtool"
+do_configure[postfuncs] += "${DO_CONFIGURE_LIBTOOL}"
+do_configure_libtool() {
+    # make the wrapper use the native mono binary instead of libtooling it
+    cat <<EOF > ${S}/runtime/mono-wrapper
+#!/bin/sh
+${BUILD_SYSROOT}/bin/mono --config ${S}/runtime/etc/mono/config "\$@"
+EOF
+}
+
+# configure make to use native mcs and mono binaries
+EXTRA_OEMAKE:machine = " \
+    EXTERNAL_MCS=${BUILD_SYSROOT}/bin/mcs \
+    EXTERNAL_RUNTIME=${BUILD_SYSROOT}/bin/mono \
+    V=1 \
+"
+
+DO_INSTALL_FUNC = "do_install_make"
+DO_INSTALL_FUNC:native  = "do_install_native"
+do_install() {
+    ${DO_INSTALL_FUNC}
+}
+
+# partly rebuilds cannot find its own binaries, so
+# help it out by updating the path
+do_install_native() {
+    export MONO_PATH=${S}/mcs/class/lib/basic
+    export PATH=$PATH:${S}/runtime/_tmpinst/bin/
+    do_install_make
+}
+
+# fix the mcs wrapper script to use the relative mono and
+# mcs binaries from the staging dir
+DO_INSTALL_MCS = ""
+DO_INSTALL_MCS:native = "do_install_mcs"
+do_install[postfuncs] += "${DO_INSTALL_MCS}"
+do_install_mcs() {
+    sed -i ${D}${bindir}/mcs \
+        -e 's!/bin/mono!mono!' \
+        -re 's!(/lib/mono/4.5/mcs.exe)!$(dirname $0)/..\1!'
+}
+
+FILES_${PN} += "${libdir} ${datadir}"
+DEPENDS_${PN} += "libm libdl libpthread libc libgcc librt libstdc++"
+RDEPENDS_${PN} += " \
+    libm \
+    libdl \
+    libpthread \
+    libc \
+    libgcc-s \
+    librt \
+    libstdc++ \
+    libmono-2.0 \
+    libmonosgen-2.0 \
+    libmonosupportw \
+    libmonoposixhelper \
+    ikvm-native \
+"
+
+AUTO_PACKAGE_LIBS = " \
+    mono-2.0 \
+    monosgen-2.0 \
+    mono-profiler-iomap \
+    mono-profiler-log \
+    mono-profiler-aot \
+    monosupportw \
+    ikvm-native \
+    monoposixhelper \
+"
+AUTO_PACKAGE_LIBS:>machine = " monoboehm-2.0"
+AUTO_PACKAGE_LIBS_DEPENDS = "libc libdl libgcc libm libpthread librt libstdc++"
+AUTO_PACKAGE_LIBS_RDEPENDS = "libc libdl libgcc libm libpthread librt libstdc++"
+
+# libmono-2.0
+LIBRARY_VERSION_${PN}-libmono-2.0 = "1.0.0"
```

Comment:
Is this the right LIBRARY_VERSION?  And if so, why is it necessary to add libmono-2.0-1.0.0 to allow-missing-soname?